### PR TITLE
Separate out arithmetic tests for datetimelike indexes

### PR DESCRIPTION
**Pull request by jbrockmendel**
As discussed... somewhere.  I'll find the reference tomorrow.

This is about as close as I could get to pure cut/paste.  There are a few notes to sprinkle in about things to clean up in a follow-up.

In a couple places this wraps some string literals to make folding easier.

### CODE REVIEW

File: pandas/tests/indexes/timedeltas/test_arithmetic.py [hunk 1]
```diff
@@ -0,0 +1,802 @@
...
+        s = Series(rng)
+        result1 = rng - pd.offsets.Hour(1)
+        result2 = DatetimeIndex(s - np.timedelta64(100000000))
+        result3 = rng - np.timedelta64(100000000)
+        result4 = DatetimeIndex(s - pd.offsets.Hour(1))
+        tm.assert_index_equal(result1, result4)
+        tm.assert_index_equal(result2, result3)
+
+
+# FIXME: This was moved from `test_tools`.  It belongs in scalar tests
```
Comment (jreback) [inline pandas/tests/indexes/timedeltas/test_arithmetic.py]: so let's move it there

File: pandas/tests/indexes/timedeltas/test_arithmetic.py [hunk 2]
```diff
@@ -0,0 +1,802 @@
...
+        rng = pd.timedelta_range('-2 days', periods=5, freq='D', name='x')
+
+        result = abs(rng)
+        exp = TimedeltaIndex(['2 days', '1 days', '0 days', '1 days',
+                              '2 days'], name='x')
+        tm.assert_index_equal(result, exp)
+        assert result.freq is None
+
+
+# FIXME: This came from `test_timedelta`, does not appear to have
```
Comment (jreback) [inline pandas/tests/indexes/timedeltas/test_arithmetic.py]: the class names came from a previous round of moving tests, you can remove/rename them if it doesn't make sense (in fact I generally *don't* like classes for organizing , and much prefer separate well-named files, changing this is out of scope for this PR). This plays nicer with pytest.

**Review by jreback — CHANGES_REQUESTED** [review]
can you show the before & after number of tests fo r the affected files (to make sure not missing any). a couple of comments.

**Comment by jbrockmendel** [comment]
Looking at tests.indexes: 3824 under PR, 3829 under master.  Removal of duplicate Julian tests accounts for the difference

**Comment by jbrockmendel** [comment]
From Travis log:

```
=================================== FAILURES ===================================
________________ TestClipboard.test_round_trip_valid_encodings _________________
self = <pandas.tests.io.test_clipboard.TestClipboard object at 0x7f8377152438>
    def test_round_trip_valid_encodings(self):
        for enc in ['UTF-8', 'utf-8', 'utf8']:
            for dt in self.data_types:
>               self.check_round_trip_frame(dt, encoding=enc)
pandas/tests/io/test_clipboard.py:136: 
[...]
>       raise AssertionError(msg)
E       AssertionError: DataFrame are different
E       
E       DataFrame shape mismatch
E       [left]:  (5, 3)
E       [right]: (2, 2)
[...]
pandas/util/testing.py:1089: AssertionError
============================ 15513 tests deselected ============================
 1 failed, 518 passed, 22 skipped, 15513 deselected, 1 xfailed in 164.49 seconds 
sys:1: ResourceWarning: unclosed <socket.socket fd=4, family=AddressFamily.AF_INET, type=2050, proto=0, laddr=('0.0.0.0', 0)>
[...]
..ValueError: Buffer dtype mismatch, expected 'Python object' but got 'long'
Exception ignored in: 'pandas._libs.lib.is_bool_array'
ValueError: Buffer dtype mismatch, expected 'Python object' but got 'long'
.................ValueError: Buffer dtype mismatch, expected 'Python object' but got 'long'
Exception ignored in: 'pandas._libs.lib.is_bool_array'
ValueError: Buffer dtype mismatch, expected 'Python object' but got 'long'
.ValueError: Buffer dtype mismatch, expected 'Python object' but got 'long'
Exception ignored in: 'pandas._libs.lib.is_bool_array'
ValueError: Buffer dtype mismatch, expected 'Python object' but got 'long'
```

No idea if the buffer mismatches are related to the TestClipboard error.  The latter has shown up a handful of times and usually goes away with re-running the CI.

**Comment by jreback** [comment]
the clipboard tests are orthogonal to this. They fail occasionally, a restart of the job usually fixes. 

However the is_bool_array is faulting somewhere in cython.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks!

**Pull request merged**